### PR TITLE
[ROCm] Unskip scaled_dot_product_attention tests on ROCm

### PR DESCRIPTION
**Pull request by alugorey**
Needle has moved quite a bit on the ROCm backend front. This PR intended to examine the tests referenced in the following issue: https://github.com/pytorch/pytorch/issues/96560

This a follow-up PR to https://github.com/pytorch/pytorch/pull/125069

unskipping the next batch of tests referenced by the aforementioned issue. No explicit changes needed for source as they worked immediately after unskipping. 

The tests previously marked with xfail have now been modified to not expect a failure iff running on ROCm as they now pass. Behavior is unchanged for them on other architectures.

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang

### CODE REVIEW

**Comment by pruthvistony** [comment]
Enabled UTs are passing.
https://github.com/pytorch/pytorch/actions/runs/9392518443/job/25867491886

**Review by pruthvistony — DISMISSED** [review]

**Review by zou3519 — DISMISSED** [review]

**Comment by jithunnair-amd** [comment]
Some examples of failures:
[ossci-raw-job-status.s3.amazonaws.com/log/26240852749](https://ossci-raw-job-status.s3.amazonaws.com/log/26240852749)
 
[ossci-raw-job-status.s3.amazonaws.com/log/26240852749](https://ossci-raw-job-status.s3.amazonaws.com/log/26240852749)

**Comment by alugorey** [comment]
Affected tests pass here: https://github.com/pytorch/pytorch/actions/runs/9555338098/job/26339365253?pr=127966

@jithunnair-amd there seems to be a rocm failure in there. I was able to reproduce locally, however, when removing the commits of this PR, the error was still encountered, thus proving that it's unrelated to this piece of work. Still, something we should investigate further.

**Comment by jithunnair-amd** [comment]
> Affected tests pass here: https://github.com/pytorch/pytorch/actions/runs/9555338098/job/26339365253?pr=127966
> 
> @jithunnair-amd there seems to be a rocm failure in there. I was able to reproduce locally, however, when removing the commits of this PR, the error was still encountered, thus proving that it's unrelated to this piece of work. Still, something we should investigate further.

That failure is also seen in trunk, so it's not related to this PR.

**Comment by jithunnair-amd** [comment]
@malfet This PR is ready to merge. ROCM CI passed. It needs a re-review because we reverted the PR to fix some unit test failures.

**Comment by alugorey** [comment]
Hello, @zou3519 Could we get someone on your team to take one last look at this?

**Comment by jithunnair-amd** [comment]
@malfet @zou3519 Can we get a re-review of this PR please?

File: test/functorch/test_ops.py [hunk 1]
```diff
@@ -351,7 +351,7 @@ def is_inplace(op, variant):
 
 vjp_fail = {
     xfail("tensor_split"),  # data_ptr composite compliance
-    # https://github.com/pytorch/pytorch/issues/96560
+    # Very minor accuracy issue on ROCm
```
Comment (malfet) [inline test/functorch/test_ops.py:354]: Why this change is needed?

**Review by malfet — APPROVED** [review]

File: test/functorch/test_ops.py (hunk 1 above)
Comment (alugorey) [reply test/functorch/test_ops.py:354]: I wanted to decouple this skip reason from the quoted issue since we triaged it and found that it was just a very minor tolerance issue that does not require immediate attention. This way we can close the aforementioned issue.
